src/notification.c

```diff
@@ -240,6 +240,136 @@ check_value_type (const char *key,
   return FALSE;
 }
 
+static void
+markup_parser_text (GMarkupParseContext  *context,
+                    const gchar          *text,
+                    gsize                 text_len,
+                    gpointer              user_data,
+                    GError              **error)
+{
+  GString *composed = user_data;
+  gchar *str = (gchar *) text;
+
+  while (*str)
```

Comment:
I really don't like this entire function (its hard to tell what it is trying to achieve) but i don't have a good suggestion either.